modules/plugin/epsg-hsql/src/test/java/org/geotools/referencing/factory/epsg/OperationFactoryTest.java

```diff
@@ -209,4 +212,158 @@ public void testTransformedDirectPosition() throws Exception {
+        int size = operations.size();
+        assertTrue(size == 1); // only one operation since they have the same datum
+
+        for (CoordinateOperation operation : operations) {
+            assertSame(sourceCRS, operation.getSourceCRS());
+            assertSame(targetCRS, operation.getTargetCRS());
+            final MathTransform transform = operation.getMathTransform();
+            TransformTestBase.assertInterfaced(transform);
+        }
+
+        // try reverse order
+        operations = opFactory.findOperations(targetCRS, sourceCRS);
+        assertEquals(size, operations.size());
+
+        for (CoordinateOperation operation : operations) {
+            assertSame(targetCRS, operation.getSourceCRS());
+            assertSame(sourceCRS, operation.getTargetCRS());
+            final MathTransform transform = operation.getMathTransform();
+            TransformTestBase.assertInterfaced(transform);
+        }
+
+    }
+
+    /**
+     * Tests findOperations method for a pair of known CRSs. 23030 (Projected) to 3034 (Projected), with different datum
+     * 
+     */
+    public void testFindOperations3() throws Exception {
+        final CRSAuthorityFactory crsFactory;
+        final CoordinateOperationFactory opFactory;
+        CoordinateReferenceSystem sourceCRS;
+        CoordinateReferenceSystem targetCRS;
+
+        crsFactory = ReferencingFactoryFinder.getCRSAuthorityFactory("EPSG", null);
+        opFactory = ReferencingFactoryFinder.getCoordinateOperationFactory(null);
+        targetCRS = crsFactory.createCoordinateReferenceSystem("EPSG:3034");
+        sourceCRS = crsFactory.createCoordinateReferenceSystem("EPSG:23030");
+
+        Set<CoordinateOperation> operations = opFactory.findOperations(sourceCRS, targetCRS);
+        int size = operations.size();
+        assertTrue(size >= 9); // at least 9 operation should be registered in the database for this CRS pair
+
+        for (CoordinateOperation operation : operations) {
+            assertSame(sourceCRS, operation.getSourceCRS());
+            assertSame(targetCRS, operation.getTargetCRS());
+            final MathTransform transform = operation.getMathTransform();
+            TransformTestBase.assertInterfaced(transform);
+        }
+
+        // try reverse order
+        operations = opFactory.findOperations(targetCRS, sourceCRS);
+        assertEquals(size, operations.size());
+
+        for (CoordinateOperation operation : operations) {
+            assertSame(targetCRS, operation.getSourceCRS());
+            assertSame(sourceCRS, operation.getTargetCRS());
+            final MathTransform transform = operation.getMathTransform();
+            TransformTestBase.assertInterfaced(transform);
+        }
+
+    }
+
+    /**
+     * Tests findOperations method for a pair of known CRSs. 4258 (Geographic2D) to 4326 (Geographic2D), with different datum
+     */
+    public void testFindOperations4() throws Exception {
```

Comment:
As an additional test for geographic to geographic try something that has a grid shift file, e.g., see this test for reference (it leverages grid shift files that you can find in this module test resources)
https://github.com/geotools/geotools/blob/e3e81c4f24b4deb3acde49948c650e85df74d850/modules/library/referencing/src/test/java/org/geotools/referencing/AbstractCRSTest.java#L475